src/cli/generator.ts

```diff
@@ -1,41 +1,632 @@
+    let segmentBlock = generateBlock(segment.body, infoFrame);
+    if (segmentResult.length !== 0) {
+        // Segment should always have results
+        segmentBlock += `\nreturn ${segmentResult.map((result) => result.name).join(', ')}`;
+    }
+    return expandToString`def ${getPythonNameOrDefault(services, segment)}(${generateParameters(
+        segment.parameterList,
+        infoFrame,
+    )}):\n${PYTHON_INDENT}${segmentBlock}`;
+};
+
+const generatePipeline = function (
+    services: SafeDsServices,
+    pipeline: SdsPipeline,
+    importSet: Map<String, ImportData>,
+): string {
+    const infoFrame = new GenerationInfoFrame(services, importSet);
+    return expandToString`def ${getPythonNameOrDefault(services, pipeline)}():\n${PYTHON_INDENT}${generateBlock(
+        pipeline.body,
+        infoFrame,
+    )}`;
+};
+
+const generateImport = function (importStmt: ImportData): string {
+    if (importStmt.declarationName === undefined && importStmt.alias === undefined) {
+        return `import ${importStmt.importPath}`;
+    } else if (importStmt.declarationName === undefined && importStmt.alias !== undefined) {
+        return `import ${importStmt.importPath} as ${importStmt.alias}`;
+    } else if (importStmt.declarationName !== undefined && importStmt.alias === undefined) {
+        return `from ${importStmt.importPath} import ${importStmt.declarationName}`;
+    } else {
+        return `from ${importStmt.importPath} import ${importStmt.declarationName} as ${importStmt.alias}`;
+    }
+};
+
+const generateImports = function (importSet: ImportData[]): string[] {
+    const qualifiedImports = Array.from(importSet)
+        .filter((importStmt) => importStmt.declarationName === undefined)
+        .sort((a, b) => a.importPath.localeCompare(b.importPath))
+        .map(generateImport);
+    const groupedImports = Object.entries(
+        group(
+            Array.from(importSet).filter((importStmt) => importStmt.declarationName !== undefined),
+            (importStmt) => importStmt.importPath,
+        ),
+    ).sort(([key1, _value1], [key2, _value2]) => key1.localeCompare(key2));
+    const declaredImports: string[] = [];
+    for (const [key, value] of groupedImports) {
+        const importedDecls =
+            value
+                ?.filter((importData) => importData !== undefined)
+                .sort((a, b) => (<string>a.declarationName).localeCompare(<string>b.declarationName))
+                .map((localValue) =>
+                    localValue.alias !== undefined
+                        ? `${localValue.declarationName} as ${localValue.alias}`
+                        : <string>localValue.declarationName,
+                ) || [];
+        declaredImports.push(`from ${key} import ${[...new Set(importedDecls)].join(', ')}`);
+    }
+    return [...new Set(qualifiedImports), ...new Set(declaredImports)];
+};
+
+const generateModule = function (services: SafeDsServices, module: SdsModule): string {
```

Comment:
Regarding the order of functions, please order them, so the calling functions are above the called functions where possible. This makes it easier to understand the program when reading it from top to bottom.